app/workers/activitypub/distribution_worker.rb

```diff
@@ -24,22 +26,42 @@ def perform(status_id)
   private
 
   def skip_distribution?
-    @status.direct_visibility? || @status.limited_visibility?
+    @status.direct_visibility?
+  end
+
+  def delegate_distribution?
+    @status.limited_visibility? && @status.thread&.limited_visibility? && !@status.thread&.account&.local?
   end
 
   def relayable?
     @status.public_visibility?
   end
 
+  def deliver_to_parent!
+    ActivityPub::DeliveryWorker.perform_async(payload, @account.id, @status.thread.account.inbox_url)
+  end
```

Comment:
Hmmm… here you're delivering to the direct parent, but if multiple people are involved, the reply might need to get passed around quite a bit to finally reach the person who knows the proper audience.

Using something like a `Group` actor that you know at each point of the thread, may make more sense.